src/ducks/sep31Send.ts

```diff
@@ -0,0 +1,309 @@
+        senderSep12Memo: sep12Fields.info.senderSep12Memo,
+        receiverSep12Memo: sep12Fields.info.receiverSep12Memo,
+        authEndpoint: tomlResponse.authEndpoint,
+        sendServer: tomlResponse.sendServer,
+        kycServer: tomlResponse.kycServer,
+      };
+    } catch (error) {
+      log.error({
+        title: error.toString(),
+      });
+
+      return rejectWithValue({
+        errorString: error.toString(),
+      });
+    }
+  },
+);
+
+interface SubmitSep31SendTransactionActionProps {
+  amount: { amount: string };
+  transaction: AnyObject;
+  sender: AnyObject;
+  receiver: AnyObject;
+}
+
+export const submitSep31SendTransactionAction = createAsyncThunk<
+  boolean,
+  SubmitSep31SendTransactionActionProps,
+  { rejectValue: RejectMessage; state: RootState }
+>(
+  "sep31Send/submitSep31SendTransactionAction",
+  async (
+    { amount, transaction, sender, receiver },
+    { rejectWithValue, getState },
+  ) => {
+    try {
+      const { secretKey } = accountSelector(getState());
+      const { data } = sep31SendSelector(getState());
+      const { pubnet } = settingsSelector(getState());
+      const networkConfig = getNetworkConfig(pubnet);
+      const {
+        token,
+        assetCode,
+        assetIssuer,
+        kycServer,
+        sendServer,
+        senderSep12Memo,
+        receiverSep12Memo,
+        fields,
+      } = data;
+
+      // Put SEP-12 fields
+      log.instruction({
+        title: "Make PUT /customer requests for sending and receiving user",
+      });
+
+      log.instruction({ title: "Form data", body: { sender, receiver } });
+
+      const putSep12FieldsResponse = await putSep12Fields({
+        fields,
+        formData: { sender, receiver },
+        secretKey,
+        senderSep12Memo,
+        receiverSep12Memo,
+        kycServer,
+        token,
+      });
+
+      // Post transaction
+      const postResponse = await postTransaction({
+        amount: amount.amount,
+        assetCode,
+        senderId: putSep12FieldsResponse.senderSep12Id,
+        receiverId: putSep12FieldsResponse.receiverSep12Id,
+        transactionFormData: transaction,
+        sendServer,
+        token,
+      });
+
+      // Poll transaction until ready
+      await pollTransactionUntilReady({
+        sendServer,
+        transactionId: postResponse.transactionId,
+        token,
+      });
+
+      // Send payment
+      await sendPayment({
+        amount: amount.amount,
+        assetCode,
+        assetIssuer,
+        receiverAddress: postResponse.receiverAddress,
+        secretKey,
+        sendMemo: postResponse.sendMemo,
+        sendMemoType: postResponse.sendMemoType,
+        networkUrl: networkConfig.url,
+        networkPassphrase: networkConfig.network,
+      });
+
+      // Poll transaction until complete
+      await pollTransactionUntilComplete({
+        sendServer,
+        transactionId: postResponse.transactionId,
+        token,
+      });
```

Comment:
We need support the non-happy-path flows where the KYC or transaction information given is invalid and the user has to update the information, see the [possible error statuses](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0031.md#pending-customer-info-update).

Reply:
Created an issue to do this later: https://github.com/stellar/nebula/issues/159